lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useRef, useContext, useEffect, useState } from 'react';
+import classnames from 'classnames';
+
+import { nxToastPropTypes, NxToastProps } from './types';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+import ToastContext from './contexts';
+import { toastTypeMap } from './toastTypeMapping';
+
+const NxToast = (props: NxToastProps) => {
+  const { toastId, className, message, type, toastContainerRef, ...otherProps } = props,
+      toastClass = toastTypeMap[type].class,
+      toastIcon = toastTypeMap[type].icon,
+      toastIconLabel = toastTypeMap[type].iconLabel,
+      [animate, setAnimate] = useState(false),
+      classes = classnames('nx-toast', 'animate', className, toastClass,
+          {'slide-in': animate},
+          {'slide-out': !animate}
+      ),
+      toastContext = useContext(ToastContext),
+      closeBtnRef = useRef<HTMLButtonElement>(null);
+
+  //Fire effect to focus on the close button of the toast that is rendered
+  useEffect(() => {
+    const closeBtn = closeBtnRef.current;
+    if (closeBtn) {
+      closeBtn.focus();
+    }
+
+    //When component is mounted, add class "animate slide-in" to trigger animation
+    setAnimate(true);
+  }, []);
+
+  //Effect to shift focus to the next toast's close button
+  //Dependency on the global toasts array
+  useEffect(() => {
+    if (toastContainerRef) {
+      //Gets the first close button of the child from the parent toast container div
+      const closeBtn = toastContainerRef.current?.querySelectorAll('.nx-btn--close')[0] as HTMLButtonElement;
+      closeBtn.focus();
+    }
+  }, [toastContext?.toasts]);
+
+  const handleCloseClick = () => {
+    setAnimate(false);
+    //Wait for slide-out animation before removing toast from DOM
+    setTimeout(() => {
+      toastContext?.removeToast(toastId);
+    }, 250);
```

Comment:
Isn't this shorter than the animation time?

Reply:
> Isn't this shorter than the animation time?

It looks like there's a `transitionend` event.  See if you can use that rather than having to duplicate the timing between the CSS and JS

Reply:
> > Isn't this shorter than the animation time?
> 
> It looks like there's a `transitionend` event. See if you can use that rather than having to duplicate the timing between the CSS and JS

We implemented transitionend here: [262a7e2](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/262a7e2b88ce5adfb12be9e3d5dbc3560d2e37d3) 

We aren't sure if this matches the code style of other components (using the forEach and addEventListener)

Reply:
> > It looks like there's a `transitionend` event. See if you can use that rather than having to duplicate the timing between the CSS and JS
> 
> We implemented transitionend here: [262a7e2](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/262a7e2b88ce5adfb12be9e3d5dbc3560d2e37d3)
> 
> We aren't sure if this matches the code style of other components (using the forEach and addEventListener)

With Andrew's help, it has transitionEnd has been implmented correctly here: [45fddad](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/45fddad331825c8ac44e9b5eb350b20b3457f413)